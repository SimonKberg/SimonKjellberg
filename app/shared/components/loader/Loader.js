import React, { Component, PropTypes } from 'react'

import withStyles from 'helpers/withStyles'
import styles from './Loader.css'

const { string } = PropTypes

class Loader extends Component {
  static propTypes = {
    text: string
  }

  static defaultProps = {
    text: 'Loading...'
  }

  render () {
    const { text } = this.props

    return (
      <div className={styles.loader}>
        <span className={styles.loaderText}>{text}</span>
      </div>
    )
  }
}

export default withStyles(styles)(Loader)
