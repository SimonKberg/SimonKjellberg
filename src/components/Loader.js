// @flow strict
import * as React from 'react'
import styled, { css, keyframes } from 'react-emotion'

const bounce = keyframes`
  0%,
  80%,
  100% {
    transform: scale(0);
  }

  40% {
    transform: scale(1);
  }
`

const Container = styled('div')`
  position: relative;
  display: block;
  margin-bottom: 1rem;
  text-align: left;
`

const point = css`
  display: inline-block;
  width: 1ch;
  height: 1ch;
  margin-right: 0.75ch;
  border-radius: 100%;
  background-color: currentColor;
  will-change: transform;
  animation: ${bounce} 1.4s infinite ease-in-out both;
`

const Point1 = styled('span')`
  ${point};
  animation-delay: -0.32s;
`

const Point2 = styled('span')`
  ${point};
  animation-delay: -0.16s;
`

const Point3 = styled('span')`
  ${point};
  animation-delay: 0;
`

const Loader = () => (
  <Container>
    <Point1 />
    <Point2 />
    <Point3 />
  </Container>
)

export default Loader
