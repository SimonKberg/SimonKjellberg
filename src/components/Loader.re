module Styles = {
  open Css;

  let bounce =
    keyframes([
      (0, [transform(scale(0.0, 0.0))]),
      (40, [transform(scale(1.0, 1.0))]),
      (80, [transform(scale(0.0, 0.0))]),
      (100, [transform(scale(0.0, 0.0))]),
    ]);

  let container =
    style([
      position(`relative),
      display(`block),
      marginBottom(rem(1.0)),
      textAlign(`left),
    ]);

  let point =
    style([
      display(`inlineBlock),
      width(ch(1.0)),
      height(ch(1.0)),
      marginRight(ch(0.75)),
      borderRadius(pct(100.0)),
      backgroundColor(`currentColor),
      unsafe("willChange", "transform"),
      animation(
        ~duration=1400,
        ~iterationCount=`infinite,
        ~timingFunction=`easeInOut,
        ~fillMode=`both,
        bounce,
      ),
    ]);

  let point1 = merge([point, style([animationDelay(-320)])]);
  let point2 = merge([point, style([animationDelay(-160)])]);
  let point3 = merge([point, style([animationDelay(0)])]);
};

[@react.component]
let make = () => {
  <div className=Styles.container>
    <span className=Styles.point1 />
    <span className=Styles.point2 />
    <span className=Styles.point3 />
  </div>;
};

let default = make;
