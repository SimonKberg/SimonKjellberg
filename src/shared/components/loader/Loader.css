
@keyframes fade {
  0% { opacity: 0.35; }
  100% { opacity: 0.75; }
}

.loader {
  position: relative;
  display: block;
  margin-bottom: 1.5rem;
  text-align: left;

  & .loaderText {
    position: absolute;
    left: 100%;
  }
}

.point {
  display: inline-block;
  width: 1ch;
  height: 1ch;
  margin-right: 0.75ch;
  border-radius: 100%;
  background-color: black;
  will-change: transform;
  animation: bounce 1.4s infinite ease-in-out both;
}

.point1 {
  composes: point;
  animation-delay: -0.32s;
}

.point2 {
  composes: point;
  animation-delay: -0.16s;
}

.point3 {
  composes: point;
  animation-delay: 0;
}

@keyframes bounce {
  0%,
  80%,
  100% { transform: scale(0); }
  40% { transform: scale(1); }
}
