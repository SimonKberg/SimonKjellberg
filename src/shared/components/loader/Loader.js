import React, { PropTypes } from 'react'

import withStyles from 'helpers/withStyles'
import Styles from './Loader.css'

export const Loader = ({ styles }) => (
  <div className={styles.loader}>
    <span className={styles.point1}></span>
    <span className={styles.point2}></span>
    <span className={styles.point3}></span>
  </div>
)

Loader.propTypes = {
  styles: PropTypes.object
}

export default withStyles(Styles)(Loader)
