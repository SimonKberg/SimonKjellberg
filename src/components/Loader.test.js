import * as React from 'react'
import renderer from 'react-test-renderer'
import Loader from './Loader.bs'

test('<Loader />', () => {
  const tree = renderer.create(<Loader />).toJSON()
  expect(tree).toMatchSnapshot()
})
