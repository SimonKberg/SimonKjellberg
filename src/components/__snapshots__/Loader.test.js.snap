// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> 1`] = `
.emotion-3 {
  position: relative;
  display: block;
  margin-bottom: 1rem;
  text-align: left;
}

.emotion-0 {
  display: inline-block;
  width: 1ch;
  height: 1ch;
  margin-right: 0.75ch;
  border-radius: 100%;
  background-color: currentColor;
  will-change: transform;
  -webkit-animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  -webkit-animation-delay: -320ms;
  animation-delay: -320ms;
}

.emotion-1 {
  display: inline-block;
  width: 1ch;
  height: 1ch;
  margin-right: 0.75ch;
  border-radius: 100%;
  background-color: currentColor;
  will-change: transform;
  -webkit-animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  -webkit-animation-delay: -160ms;
  animation-delay: -160ms;
}

.emotion-2 {
  display: inline-block;
  width: 1ch;
  height: 1ch;
  margin-right: 0.75ch;
  border-radius: 100%;
  background-color: currentColor;
  will-change: transform;
  -webkit-animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  animation: animation-1kr0zyd 1400ms ease-in-out 0ms infinite normal both running;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

<div
  className="emotion-3"
>
  <span
    className="emotion-0"
  />
  <span
    className="emotion-1"
  />
  <span
    className="emotion-2"
  />
</div>
`;
